apps/api/src/schema/programs.ts

```diff
@@ -1,6 +1,28 @@
 import { z } from "@hono/zod-openapi";
 
 const programIdBase = z.string({ required_error: "programId is required" });
+const majorIdBase = z.string();
+
+export const majorQuerySchema = z.object({
+  majorId: majorIdBase.optional().openapi({
+    description: "The ID of the specific major being requested for",
+    example: "BA-163",
+  }),
+});
+
+export const minorQuerySchema = z.object({
+  id: z.string().optional().openapi({
+    description: "The ID of the specific minor being requested for",
+    example: "49A",
+  }),
+});
+
+export const specializationQuerySchema = z.object({
+  majorId: majorIdBase.optional().openapi({
+    description: "The ID of a major to query its associated specializations",
```

Comment:
Also a bit strange, try e.g. "Only fetch specializations associated with the major with this ID, if provided"